lib/Yield/PTML.cpp

```diff
@@ -264,22 +265,72 @@ class InstructionPrefixManager {
   }
 };
 
+struct StatementGraphNode {
+  const yield::BasicBlock *Block;
+
+  StatementGraphNode(const yield::BasicBlock &BB, const MetaAddress &) :
+    Block(&BB) {}
+};
+using StatementGraph = GenericGraph<ForwardNode<StatementGraphNode>>;
+
+template<>
+struct yield::StatementTraits<StatementGraph::Node *> {
+
+  using StatementType = const MetaAddress;
+
+  // TODO: switch to `std::optional` after updating to c++26.
+  using LocationType = const llvm::SmallVector<MetaAddress, 1>;
+
+  static auto getStatements(const StatementGraph::Node *Node) {
+    constexpr static SortedVector<yield::Instruction> Empty{};
+    return (Node->Block->ID().isValid() ? Node->Block->Instructions() : Empty)
+           | std::views::transform([](auto &&R) { return R.Address(); });
```

Comment:
Why did you choose to have `StatementType` to be a `MetaAddress` as opposed to a `yield::Instruction`?
If we ever add additional methods to this trait it will probably make sense to have `yield::Instruction`.

Reply:
My thinking was simple: being extra explicit makes it easier to understand what's going on - this way there's no doubt that it never uses anything about the instruction but its address - because it _doesn't_ have anything in the first place.

If we ever need more information about the instruction, it's pretty easy to change the `StatementType`.

Reply:
It's not too bad, but I think it kinda spoils the point of the trait, i.e., being able to operate on an opaque data structure, which in this case is a `yield::Instruction`. At this point one could wonder why not just have a single function that returns a vector of set of addresses.

No biggie though, up to you.

Reply:
Done.